DigitalLearningSolutions.Web/Services/UserService.cs

```diff
@@ -393,25 +416,55 @@ bool changeMadeBySameUser
                     delegateDetailsData.Answer4,
                     delegateDetailsData.Answer5,
                     delegateDetailsData.Answer6,
-                    detailsLastChecked
+                    currentTime
                 );
             }
 
-            UpdateUserDetails(editAccountDetailsData, changeMadeBySameUser, detailsLastChecked);
+            UpdateUserDetails(editAccountDetailsData, isPrimaryEmailUpdated, changeMadeBySameUser, currentTime);
 
-            userDataService.SetCentreEmail(
-                editAccountDetailsData.UserId,
-                centreId,
-                centreEmail,
-                changeMadeBySameUser ? (DateTime?)null : clockUtility.UtcNow
-            );
+            if (isCentreEmailUpdated)
+            {
+                var emailVerified =
+                    changeMadeBySameUser &&
+                    !emailVerificationDataService
+                        .AccountEmailIsVerifiedForUser(

```

Comment:
I think this is maybe _slightly_ easier to read if `emailVerificationDataService` and `AccountEmailIsVerifiedForUser` are on the same line